daft/catalog/__iceberg.py

```diff
@@ -110,7 +110,7 @@ def list_namespaces(self, pattern: str | None = None) -> list[Identifier]:
 
     def list_tables(self, pattern: str | None = None) -> list[str]:
         """List tables under the given namespace (pattern) in the catalog, or all tables if no namespace is provided."""
-        return [".".join(tup) for tup in self._inner.list_tables(pattern or ())]
+        return [".".join(tup) for tup in self._inner.list_tables(pattern or "")]
```

Comment:
I believe the pyiceberg logic is to use an empty tuple. If you do an empty string, it's a non-null value. The issue I believe is that the Iceberg REST ListTables API requires a namespace. Let me collect some links, but the empty tuple was deliberate.

Reply:
Iceberg catalog behavior may slightly differ across implementations, Iceberg REST requires a namespace since it's the actual API endpoint URL. https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L506-L516